src/nat/builder/intermediate_step_manager.py

```
@@ -46,11 +48,19 @@ class IntermediateStepManager:
     Manages updates to the NAT Event Stream for intermediate steps
     """
 
+    # Class-level tracking for debugging and monitoring
+    _instance_count: ClassVar[int] = 0
+    _active_instances: ClassVar[set[weakref.ref]] = set()
+
     def __init__(self, context_state: "ContextState"):  # noqa: F821
```

Ruff (0.14.2)

Unused `noqa` directive (unused: `F821`)

Remove unused `noqa` directive

(RUF100)